### PR TITLE
add security IT

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
Add security IT.
`./gradlew integTest -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername="docker-cluster" -Dhttps=true -Duser=admin -Dpassword=admin --tests="org.opensearch.ml.rest.SecureMLRestIT"`
 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        createRoleMapping(indexSearchAccessRole, new ArrayList<>(Arrays.asList(mlFullAccessUser)));
+
+        ingestIrisData(irisIndex);
+        searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.query(new MatchAllQueryBuilder());
+        searchSourceBuilder.size(1000);
+        searchSourceBuilder.fetchSource(new String[] { "petal_length_in_cm", "petal_width_in_cm" }, null);
+    }
+
+    @After
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java:94]: Just realized this annotation exists. Should we add this to clean up all the data in each other integration tests? That might fix the flaky request count problem.

**Review by Zhangxunmt — APPROVED** [review]
Do we want to add one more test for accessing those ml system indices like model and task?

**Comment by ylwu-amzn** [comment]
> 

Yes, will add more test cased in next PRs.

File: plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java (hunk 1 above)
Comment (ylwu-amzn) [reply plugin/src/test/java/org/opensearch/ml/rest/SecureMLRestIT.java:94]: For the same cluster, the total request count will not decrease, unless we restart the cluster. The total request is some value cached in memory not in index. So cleanup data won't help

**Pull request merged**